warmup/bootstrapper.py

```diff
@@ -70,43 +70,68 @@ def _mean(data):
     return math.fsum(data) / float(len(data))
 
 
-def _bootstrap_means(steady_segments_all_pexecs):
+def _bootstrap_means_lowq(steady_segments_all_pexecs):
     # How many bootstrap samples do we need from each pexec? We want at least
     # BOOTSTRAP_ITERATIONS samples over all. If we want 100,000 samples in total
     # and we have 30 pexecs, we need 3333 samples from each pexec. In total we
     # will have 3333 * 30 bootstrapped samples, and 3333 * 30 == 99990. So, we
     # add a 1 here to ensure that we end up with >= BOOTSTRAP_ITERATIONS samples.
-    n_resamples = int(math.floor(BOOTSTRAP_ITERATIONS / len(steady_segments_all_pexecs))) + 1
+    n_resamples = int(math.floor(BOOTSTRAP_ITERATIONS_LOWQ / len(steady_segments_all_pexecs))) + 1
     means = list()  # Final list of BOOTSTRAP_ITERATIONS resamples.
 
     for segments in steady_segments_all_pexecs:  # Iterate over pexecs.
         for _ in xrange(n_resamples):
             num_samples = 0
 
-            # Note that summing into a float like this does cause rounding errors, but for 100,000
-            # bootstrap iterations that error is only apparent at the 9th decimal point, which is
-            # acceptable in our case. Roughly speaking, each factor of 10 bigger that
-            # BOOTSTRAP_ITERATIONS becomes, the error becomes 1 decimal place "worse",
-            # so if you ever crank BOOTSTRAP_ITERATIONS to a mammoth value, keep this in mind.
+            # Note that summing into a float like this does cause rounding
+            # errors, but for 10,000 bootstrap iterations and values in the
+            # 0.0-1.0 range, that error is only apparent at the 10th decimal
+            # point. However, the error increases as the bootstrap iterations
+            # and/or timings increase above: roughly speaking the error becomes
+            # 1 decimal place worse for each order of magnitude bigger the
+            # bootstrap iterations and/or timings become.
             sample_sum = 0.0
-
             for seg in segments:
                 seg_len = len(seg)
                 num_samples += seg_len
                 for _ in xrange(seg_len):
                     sample_sum += seg[int(random.random() * seg_len)]
 
             means.append(sample_sum / float(num_samples))
-    assert len(means) >= BOOTSTRAP_ITERATIONS
+    assert len(means) >= BOOTSTRAP_ITERATIONS_LOWQ
+    return means
+
+def _bootstrap_means_highq(steady_segments_all_pexecs):
```

Comment:
Did you put the high quality code in a separate function for performance purposes? To avoid repeated "am I hi quality" conditionals?

Reply:
The latter.

Reply:
As long as it was a concious decision I don't mind either way. I was expecting one function with conditionals, and perhaps assumed you were doing something cleverer than you were :)

(I moved this comment from above, as I originally posted in the wrong place, sorry)

Reply:
Note that the two functions have fairly different bodies: I think that making various bits conditional would make it hard to read.